src/sentry/nodestore/bigtable/backend.py

```diff
@@ -32,6 +33,38 @@
 _connection_cache = {}
 
 
+def _compress_data(orig_data, data, compression):
+    flags = 0
+
+    if callable(compression):
+        compression = compression(orig_data)
+
+    if compression == "zstd":
+        flags |= BigtableNodeStorage._FLAG_COMPRESSED
```

Comment:
Why not add a new flag? Instead of sniffing bytes to determine if it's zstd or zlib? Seems much better to just add a new flag and check that.

Reply:
Is there a compelling reason for either way? I am doing byte-sniffing since those bytes are part of the zstd spec (this is *not* like character encoding detection) and I don't like to store the same information twice in the DB

Reply:
as arpad pointed out it could be easier to query for this way, I'll just change it